### PR TITLE
Jira Feature Collector Refactor

### PR DESCRIPTION
**Pull request by amp5208**
I have refactored the entire JiraFeatureCollector based on @jdamick and PMD automated feedback.  I have also re-wired the whole transformation and client layer to work with the Jira Atlasssian JRJC library, which ultimately makes heavy use of Jira's domain object model (as opposed to raw HTTP calls and JSON transformation).

Please fully review my code and provide feedback.  I would also explicitly like feedback in a section that I could not remove all PMD errors on:  <strong>StoryDataClientImpl.java, line 84-86</strong>.  Although this section is working and tested and has been refactored as stated above, I cannot reduct overall complexity of the method, because full domain object model translation of hierarchical issues is inherently complex.

<strong>Note:</strong> Please do not fully merge this code until internal functional testing passes early next week.  I will post here when functional tests have passed.

### CODE REVIEW

File: jira-feature-collector/src/main/java/com/capitalone/dashboard/client/project/ProjectDataClientImpl.java [hunk 1]
```diff
@@ -45,79 +42,75 @@ public ProjectDataClientImpl(FeatureSettings featureSettings,
...
-				} catch (Exception e) {
-					LOGGER.error("Unexpected error caused when attempting to save data\nCaused by:\n"
-							+ e.getMessage()
-							+ " : "
-							+ e.getCause()
-							+ "\n"
-							+ Arrays.toString(e.getStackTrace()));
 				}
 			}
 		} catch (Exception e) {
```
Comment (amitmawkin) [inline jira-feature-collector/src/main/java/com/capitalone/dashboard/client/project/ProjectDataClientImpl.java]: Please use specific Exceptions to catch, It is considered a bad practice if you are using Parent Exception class to catch all exception, plus it becomes hard to debug too .

**Comment by amp5208** [comment]
I have fixed the exception cases you noted above as well as some bug fixes related to the use of JRJC and closing connections.  Please validate.

However, the Travis CI build appears to be failing because of test failurs in Core... I did not touch core, as you can see in my commit log.  I am going to try and fetch the upstream content to ensure that newest merges to upstream master did not affect the travis CI build from my older copy.

**Comment by amp5208** [comment]
Fetched upstream successfully and merged newest content down to my fork.  Travis CI issues should now be resolved.

**Comment by amp5208** [comment]
Thank you for merging - this has also passed functional testing (noting retroactively for completeness).

**Pull request merged**